f3dex2.s

```diff
@@ -1648,35 +1778,33 @@ no_z_buffer:
     j       check_rdp_buffer_full   // eventually returns to $ra, which is next cmd, second tri in TRI2, or middle of clipping
      sdv    $v18[8], 0x0000($1)     // Store S, T, W texture coefficients (integer)
 
-vtxPtr equ $25
-endVtxPtr equ $24
 G_CULLDL_handler:
-    lhu     vtxPtr, (vertexTable)(cmd_w0)   // load start vertex address
-    lhu     endVtxPtr, (vertexTable)(cmd_w1) // load end vertex address
+    lhu     cmd_w0, (vertexTable)(cmd_w0)     // load start vertex address
+    lhu     cmd_w1_dram, (vertexTable)(cmd_w1_dram) // load end vertex address
     addiu   $1, $zero, (CLIP_NX | CLIP_NY | CLIP_PX | CLIP_PY | CLIP_FAR | CLIP_NEAR)
-    lw      $11, VTX_CLIP(vtxPtr)           // read clip flags from vertex
+    lw      $11, VTX_CLIP(cmd_w0)             // read clip flags from vertex
 culldl_loop:
     and     $1, $1, $11
-    beqz    $1, run_next_DL_command         // Some vertex is on the screen-side of all clipping planes; have to render
-     lw     $11, (vtxSize + VTX_CLIP)(vtxPtr) // next vertex clip flags
-    bne     vtxPtr, endVtxPtr, culldl_loop  // loop until reaching the last vertex
-     addiu  vtxPtr, vtxPtr, vtxSize         // advance to the next vertex
-    j       G_ENDDL_handler                 // If got here, there's some clipping plane where all verts are outside it; skip DL
+    beqz    $1, run_next_DL_command           // Some vertex is on the screen-side of all clipping planes; have to render
+     lw     $11, (vtxSize + VTX_CLIP)(cmd_w0) // next vertex clip flags
+    bne     cmd_w0, cmd_w1_dram, culldl_loop  // loop until reaching the last vertex
```

Comment:
Overall I like the new naming, but there are some spots where the new names are less descriptive than the old ones like this one. I think local names for global registers is fine as long as there's a clear notice when the register is renamed.

Reply:
I've changed this set back--you're right, this is a large enough use of them for a different purpose to be worth using local names.